frontend/ui/widget/iconbutton.lua

```diff
@@ -125,7 +125,6 @@ function IconButton:onHoldIconButton()
 end
 
 function IconButton:onSetDimensions(new_dimen)
-    self.dimen = new_dimen
```

Comment:
This does seem like a potentially somewhat odd thing to do (I think SetDimensions is about the screen size?) but could you explain what's behind removing this line?

In any case it seems to me that the entire thing should be removed if it's just an empty function.

Reply:
This event:
https://github.com/koreader/koreader/blob/1d1b005b1ad98d3dad2703d94288e4245a63e265/frontend/apps/filemanager/filemanager.lua#L320
triggered and changing dimensions in IconButton
https://github.com/koreader/koreader/blob/1d1b005b1ad98d3dad2703d94288e4245a63e265/frontend/ui/widget/iconbutton.lua#L127

This causes that tap and hold does not work on the "plus icon" because they are overwritten by TAP_ZONE_MENU.

See that ButtonWidget does not have the onSetDimensions() method, so I think it can be removed from IconButton.

Reply:
Alright, so let's just take out the whole function then. Having it empty doesn't really seem to serve a purpose. :-)